spatialmath/geo_obstacle.go

```diff
@@ -126,25 +127,16 @@ func GetCartesianDistance(p, q *geo.Point) (float64, float64) {
 }
 
 // GeoPoseToPose returns the pose of point with respect to origin.
-func GeoPoseToPose(point, origin GeoPose) Pose {
+func GeoPoseToPose(point, origin *GeoPose) Pose {
 	localBearing := origin.Location().BearingTo(point.Location())
 	absoluteBearing := localBearing - origin.Heading()
 
 	latDist, lngDist := GetCartesianDistance(point.Location(), origin.Location())
 	v := r3.Vector{X: latDist * 1e-6, Y: lngDist * 1e-6, Z: 0}
 
-	nullIsland := geo.NewPoint(0, 0)
-
-	gp := nullIsland.PointAtDistanceAndBearing(v.Norm(), absoluteBearing)
-
-	position := GeoPointToPoint(gp, nullIsland)
-
-	headingChange := math.Mod(origin.Heading()-point.Heading(), 360)
-	if headingChange < 0 {
-		headingChange += 360
-	}
-
-	return NewPose(position, &OrientationVectorDegrees{OZ: 1, Theta: headingChange})
+	newPoint := origin.Location().PointAtDistanceAndBearing(v.Norm(), absoluteBearing)
+	headingChange := normalizeAngle(origin.Heading() - point.Heading())
+	return NewPose(GeoPointToPoint(newPoint, origin.Location()), &OrientationVectorDegrees{OZ: 1, Theta: headingChange})
```

Comment:
do we not need to convert from `LHR` (what headingChange is in, to RHR (what Theta is in)?

Reply:
I'm pretty sure the answer lies within the order of subtraction making it right handed.  Usually you would want to subtract out the origin frame and this does it in the reverse order which I think flips the sign effectively, making it right handed.  I can add a comment explaining this but could you take a look at this again and see if my explanation makes sense?

Reply:
cc: @nfranczak

Reply:
> within the order of subtraction making it right handed. 

I agree. Also drew up a quick example.
![IMG_7947](https://github.com/viamrobotics/rdk/assets/46872531/9ec77afe-0ece-4d28-9c44-12a54ce34895)